src/ore/src/cast.rs

```diff
@@ -82,6 +82,39 @@ cast_from!(isize, i64);
 #[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
 cast_from!(isize, i128);
 
+/// A trait for reinterpreting casts.
+///
+/// `ReinterpretCast` is like `as`, but it allows the caller to be specific about their
+/// intentions to reinterpreting the bytes from one type to another. For example, if we
+/// have some `u32` that we want to use as the return value of a postgres function, and
+/// we don't mind converting large unsigned numbers to negative signed numbers, then
+/// we would use `ReinterpretCast<i32>`.
+///
+/// `ReinterpretCast` should be preferred to the `as` operator, since it explicitly
+/// conveys the intention to reinterpret the type.
+pub trait ReinterpretCast<T> {
+    /// Performs the cast.
+    fn reinterpret_cast(from: T) -> Self;
+}
+
+macro_rules! reinterpret_cast {
+    ($from:ty, $to:ty) => {
+        paste! {
```

Comment:
not sure where the `paste!` macro is needed here...

Reply:
I honestly just copied the macro above, so I'm not sure. Thanks for calling this out, I have a PR to remove it: https://github.com/MaterializeInc/materialize/pull/16731

Reply:
Yea, you kind of need it there because of this line:
```rust
[< $from _to_ $to >]
```
which synthesizes the name of the generated function from both the `$from` and the `$to` identifiers.